cli/exchange/service.go

```diff
@@ -476,14 +476,38 @@ func ServiceUpdatePolicy(org string, credToUse string, service string, jsonFileP
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, "Incorrect policy format in file %s: %v", jsonFilePath, err)
 	}
 
+	// Set default built in properties before publishing to the exchange
+	properties := policyFile.Properties
+	for i, property := range properties {
+		// remove the {"name":"prop1", "value":"value1"} from the generated service.policy.json template and set defualt built in properties
+		if property.Name == "prop1" {
+			properties = append(properties[:i], properties[i+1:]...)
+		}
+
+	}
+
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_URL, "$SERVICE_NAME"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_NAME, "$SERVICE_NAME"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ORG, "$HZN_ORG_ID"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_VERSION, "$SERVICE_VERSION"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ARCH, "$ARCH"), false)
```

Comment:
You need to get the service and replace SERVICE_NAME, HZN_ORG_ID, SERVICE_VERSION and ARCH with real values from the service.

Reply:
you mean something like `properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_NAME, getEnvs("$SERVICE_NAME")), false)`?